### ISSUE TITLE
Metrics hold previous state

### ISSUE THREAD

**Issue by malhussan** [marcinbudny/ravendb_exporter#7]
Hi @marcinbudny, 

There seems to be an issue with my last PR https://github.com/marcinbudny/ravendb_exporter/pull/6, that is: metric `ravendb_database_tasks` keeps old states. For example, if a task **connection_status** were to change from `NotActive` to `Active`, this results in the following series:

```
ravendb_database_tasks{connection_status="NotActive",database="Demo",type="Subscription"} 2
ravendb_database_tasks{connection_status="Active",database="Demo",type="Subscription"} 2
```

which is not very helpful as it doesn't represent the current state. This is the only metric with multiple labels, so I assume it would also affect others should they have additional labels.

Looking at the code, a quick fix would be adding a reset in `collectPerDatabaseGauge`/`collectPerDatabaseCounter` functions

```diff
func collectPerDatabaseGauge(stats *stats, vec *prometheus.GaugeVec, collectFunc func(*dbStats) []metricInfo, ch chan<- prometheus.Metric) {
+	vec.Reset()
	for _, dbs := range stats.dbStats {
		metricInfos := collectFunc(dbs)

		for _, metricInfo := range metricInfos {
			vec.With(metricInfo.Labels).Set(metricInfo.Value)
		}
	}
	vec.Collect(ch)
}
```

Any ideas or suggestions?

**Comment by marcinbudny** [comment]
If this approach works, then let's apply it.

A better way would be to recreate metrics each time collection runs, similar to the approach in my [EventStoreDB exporter](https://github.com/marcinbudny/eventstore_exporter/blob/master/internal/collector/collector.go#L169). This would solve issues with concurrent metric requests.

However it would require a major refactoring and also bump of prometheus lib (here, very old version is used). So we can use the quick fix that you proposed for now.

**Comment by malhussan** [comment]
@marcinbudny thanks for the quick reply. https://github.com/marcinbudny/ravendb_exporter/pull/8 opened

**Comment by marcinbudny** [comment]
Fix published in 0.5.1
